src/main/java/seedu/address/model/expenditure/Description.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.model.expenditure;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a description of an expense in the application.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDescription(String)}
+ */
+public class Description {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Descriptions should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * The first character of the description must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final String fullDescription;
+
+    /**
+     * Constructs a {@code Description}.
+     *
+     * @param description A valid description.
+     */
+    public Description(String description) {
+        requireNonNull(description);
+        checkArgument(isValidDescription(description), MESSAGE_CONSTRAINTS);
+        fullDescription = description;
+    }
+
+    /**
+     * Returns true if a given string is a valid description.
+     */
+    public static boolean isValidDescription(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+
+    @Override
+    public String toString() {
+        return fullDescription;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Description // instanceof handles nulls
+                && fullDescription.equals(((Description) other).fullDescription)); // state check
+    }
+
+    @Override
+    public int hashCode() {
+        return fullDescription.hashCode();
+    }
+}
```

Comment:
```suggestion
}

```

I think there's 2 `Description.java` without newline. The other one was fixed

Reply:
Oh yes.. I forgot I have not deleted the other `Description.java` file, thanks!